### ISSUE TITLE
Only the right side is rendered normally

### ISSUE THREAD

**Issue by ZhanYang-Feng** [aren227/unity-fluid-simulation#1]
frame debugger - first draw mesh（In the first pass I made the ps return red）
![4CL9SLN)X929RG71M5YE4`3](https://user-images.githubusercontent.com/67536651/220058230-8d782f31-c726-44d6-a461-5ca592490b34.png)

but when i change the 1st pass in `spheres.shader` 
![YYM@G~(CY}7Q%_AQ3FML4PY](https://user-images.githubusercontent.com/67536651/220058690-d7ec49c5-51af-4953-90c8-aa277e48a989.png)
The screen image becomes like this
![_LBG(7HJB(JD~V KVKDXTUR](https://user-images.githubusercontent.com/67536651/220059037-a16354d0-0fe6-4ced-bb70-e07bcd70271e.png)
So I guess there is something wrong with the ellip/pca calculation ?(I am not familiar with this surfaces rescontruction)

**Comment by aren227** [comment]
Hi, ZhanYang-Feng!
Well, It works quite well on my machine.. 🤔 
![image](https://user-images.githubusercontent.com/14082448/220123119-5c2c2ee4-a4b6-4bf4-b67c-1cf172fdd451.png)
If you have any changes other than that, please tell me about them.
I don't know why some artifacts exist in the second image. (Black dots and noisy edges.. I guess the particle mesh is not sphere?)

`float3 worldPos = localPos + spherePos;`
This is totally fine. One minor problem is that all particles will look the same as per-particle rotation/scale are not applied.
So.. maybe the `principle` buffer was somewhat corrupted? `ellip` is just a rotation & scale matrix.

I just printed out `principle` buffer values as color output. (just added TEXCOORD0 to `v2f`)
1. `principle[id*4+0]`
![image](https://user-images.githubusercontent.com/14082448/220131124-52ad7dfc-594f-47d6-a5e6-0ccf6ff9bdc2.png)
2. `principle[id*4+1]`
![image](https://user-images.githubusercontent.com/14082448/220131267-3062ede2-5504-457d-8d03-1fc68c5744e5.png)
3. `principle[id*4+2]`
![image](https://user-images.githubusercontent.com/14082448/220131454-352607c1-6698-485e-8d1e-b72408dfbbdc.png)

**Comment by ZhanYang-Feng** [comment]
Thank you for your reply. As you say, the problem is probably `principle` buffer related. 
I also tried this print, for example, `principle[id*4+1]` :
![image](https://user-images.githubusercontent.com/67536651/220138905-951fc687-8fa6-40f7-a157-123f80a7a406.png)
This does look like the latest version of the pca module.
![image](https://user-images.githubusercontent.com/67536651/220145076-d178343e-5cad-4a6c-82ac-3a7d0ccdd151.png)
I don't know the exact reason for this problem yet. I will read the anisotropic kernels paper and check the pca module of this compute shader. If I find something, I'll let you know.
One last thing, your project is really cool!

**Comment by ShadoWKamille** [comment]
I have the same bug too.
I set numParticles to 8192 in "Solver" for debug and in "CalcPCA" kernel a lot of nan matrix are calculated in "principle" buffer.
I found these nan are happened when get wrong cov, (0, 0, 0) are returned from buffer, but the data is not (0, 0, 0).
So I move CalcPCA to another ComputeShader and modify "mean" and "cov" to StructuredBuffer, not RWStructuredBuffer.
After that, I get the right rendering result.
But I have no idea why get the wrong cov data.
